backend/main/src/Group/Domain/Entities/Container.ts

```diff
@@ -0,0 +1,99 @@
+import { Food } from "src/Group/Domain/Entities/Food";
+import { AggregateRoot, Err, Identifier, Ok, Result } from "src/Shared";
+import { DomainObjectError } from "src/Shared";
+
+interface IContainerProps {
+	name: string;
+	foods: Food[];
+}
+
+/**
+ * This class is container class.
+ * The name of container must be shorter than equal to 50 and greater than equal 1.
+ */
+export class Container extends AggregateRoot<string, IContainerProps> {
+	static create(
+		id: ContainerId,
+		props: IContainerProps,
+	): Result<Container, ContainerDomainError> {
+		if (props.name.length < 1) {
+			return Err(new ContainerDomainError("Container name is too short"));
+		}
+		if (props.name.length > 50) {
+			return Err(new ContainerDomainError("Container name is too long"));
+		}
+		return Ok(
+			new Container(id, {
+				...props,
+			}),
+		);
+	}
+
+	get name(): string {
+		return this.props.name;
+	}
+
+	get foods(): Food[] {
+		return this.props.foods;
+	}
+
+	/**
+	 * Change the name of container.
+	 * @param name
+	 */
+	public changeName(name: string): Result<Container, ContainerDomainError> {
+		return Container.create(this.id, {
```

Comment:
I think this one needs some restrictions. Like 1 < string.length < 255.

Reply:
I changed the length limitation of the name in the constructor.
It doesn't need to write validation here because the create method returns Err if it violates the restriction. I added a test case to ensure the changeName method returns Err to an invalid name.

Reply:
You are correct.